internal/attestation/measurements/measurements.go

```diff
@@ -130,41 +153,73 @@ func (m Measurements) EqualTo(other Measurements) bool {
 		return false
 	}
 	for k, v := range m {
-		if !bytes.Equal(v, other[k]) {
+		otherExpected := other[k].Expected
+		if !bytes.Equal(v.Expected[:], otherExpected[:]) {
+			return false
+		}
+		if v.Enforced != other[k].Enforced {
 			return false
 		}
 	}
 	return true
 }
 
-// MarshalYAML overwrites the default behaviour of writing out []byte not as
+// MarshalYAML overwrites the default behaviour of writing out [32]byte not as
 // single bytes, but as a single base64 encoded string.
 func (m Measurements) MarshalYAML() (any, error) {
-	base64Map := make(map[uint32]string)
+	base64Map := make(map[uint32]b64Measurement)
 
 	for key, value := range m {
-		base64Map[key] = base64.StdEncoding.EncodeToString(value[:])
+		base64Map[key] = b64Measurement{
+			Expected: base64.StdEncoding.EncodeToString(value.Expected[:]),
+			Enforced: value.Enforced,
+		}
 	}
 
 	return base64Map, nil
 }
 
-// UnmarshalYAML overwrites the default behaviour of reading []byte not as
+// UnmarshalYAML overwrites the default behaviour of reading [32]byte not as
 // single bytes, but as a single base64 encoded string.
 func (m *Measurements) UnmarshalYAML(unmarshal func(any) error) error {
-	base64Map := make(map[uint32]string)
+	base64Map := make(map[uint32]b64Measurement)
 	err := unmarshal(base64Map)
 	if err != nil {
-		return err
+		// If the unmarshal fails, try to unmarshal as a map of uint32 to string.
+		typeErr := &yaml.TypeError{}
+		if errors.As(err, &typeErr) {
+			base64SimpleMap := make(map[uint32]string)
+			err = unmarshal(base64SimpleMap)
+			if err != nil {
+				return err
+			}
+
+			for key, value := range base64SimpleMap {
+				base64Map[key] = b64Measurement{
+					Expected: value,
+					Enforced: false,
+				}
+			}
+		} else {
+			return err
+		}
 	}
 
 	*m = make(Measurements)
 	for key, value := range base64Map {
-		measurement, err := base64.StdEncoding.DecodeString(value)
+		measurement, err := base64.StdEncoding.DecodeString(value.Expected)
 		if err != nil {
 			return err
 		}
-		(*m)[key] = measurement
+
+		if len(measurement) != 32 {
+			return fmt.Errorf("invalid measurement at key %d: invalid length: %d", key, len(measurement))
+		}
```

Comment:
Maybe we could wrap this error and catch it to make it user-friendlier?
But we cannot do it in the go validator unfortunately due to the fixed size byte array.